### PR TITLE
Ported new profile for BOT from Stanford Pro profile

### PR DESCRIPTION
**Pull request by mdyoung3**
# READY FOR REVIEW

# Summary
- Port profile over for Trustees website.

# Review By (Date)
01/11

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Urgency
- Normal

# Review Tasks
1. Compare to CAW profile: https://github.com/SU-SWS/caw_profile/pull/1/files
2. Optional: Run in Drupal site.
Note: We can write the tests after we get site going. 

## Setup tasks and/or behavior to test

1. Check against what was done for CAW: https://github.com/SU-SWS/caw_profile/pull/1/files

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- [JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)](https://stanfordits.atlassian.net/browse/ADOPTD8-492)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: modules/trustee_profile_helper/ trustee_profile_helper.module [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * @file
+ * Contains caw_profile_helper.module.
```
Comment (joegl) [inline modules/trustee_profile_helper/ trustee_profile_helper.module]:
```suggestion
 * Contains trustee_profile_helper.module.
```

@mdyoung3 It also looks like there's possibly a space at the beginning of file name? 

`trustee_profile_helper/ trustee_profile_helper.module`

File: modules/trustee_profile_helper/trustee_profile_helper.info.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+name: 'Board of Trustee Profile Helper'
+type: module
+description: 'Add on module and functionality for CAW.'
+core_version_requirement: ^8.8 || ^9
+package: 'trustee'
```
Comment (joegl) [inline modules/trustee_profile_helper/trustee_profile_helper.info.yml]:
```suggestion
package: 'trustee'

```

File: trustee_profile.info.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
-name: 'Stanford Profile'
-description: 'Jumpstart Website Profile'
+name: 'Trustee Profile'
+description: 'Jumpstart Website Profile for the Board of Trustee website'
```
Comment (joegl) [inline trustee_profile.info.yml]:
```suggestion
description: 'Jumpstart Website Profile for Board of Trustees Intranet sites'
```

**Review by joegl — CHANGES_REQUESTED** [review]
@mdyoung3 This is lookin good just a few small changes/suggestions.

File: modules/trustee_profile_helper/trustee_profile_helper.info.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+name: 'Board of Trustee Profile Helper'
+type: module
+description: 'Add on module and functionality for CAW.'
```
Comment (joegl) [inline modules/trustee_profile_helper/trustee_profile_helper.info.yml]:
```suggestion
description: 'Add on module and functionality for Board of Trustees.'
```

File: modules/trustee_profile_helper/ trustee_profile_helper.module (hunk 1 above)
Comment (mdyoung3) [reply modules/trustee_profile_helper/ trustee_profile_helper.module]: Fixed

**Review by joegl — CHANGES_REQUESTED** [review]
Looks great! Looks like this filename still has a space in it though (unless that's just me?): `modules/trustee_profile_helper/ trustee_profile_helper.module`

trustee-space

File: modules/trustee_profile_helper/trustee_profile_helper.module [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * @file
+ * Contains caw_profile_helper.module.
```
Comment (joegl) [inline modules/trustee_profile_helper/trustee_profile_helper.module]:
```suggestion
 * Contains trustee_profile_helper.module.
```

**Review by joegl — APPROVED** [review]
This LGTM but I'd wait for Mike to peek at it before merging.

File: config/sync/core.extension.yml [hunk 6]
```diff
@@ -191,12 +191,12 @@ module:
   config_pages_overrides: 10
   externalauth: 10
   menu_link_weight: 10
-  stanford_profile_helper: 10
```
Comment (pookmish) [inline config/sync/core.extension.yml:194]: You'll want to keep using the `stanford_profile_helper`.

File: trustee_profile.install [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * @file
+ * trustee_profile.install
+ */
+
+/**
+ * Save the system pages from the original config values into state.
+ */
+function trustee_profile_update_8002() {
```
Comment (pookmish) [inline trustee_profile.install]: You can remove the function

File: trustee_profile.post_update.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    'trustee_profile_post_update_8014' => '8.x-2.9',
+    'trustee_profile_post_update_8015' => '8.x-2.9',
+  ];
+}
+
+/**
+ * Disable the core search module.
+ */
+function trustee_profile_post_update_8200(){
+  \Drupal::service('module_installer')->uninstall(['search']);
```
Comment (pookmish) [inline trustee_profile.post_update.php]: You can remove these since this profile is new.

**Review by pookmish — CHANGES_REQUESTED** [review]

**Comment by pookmish** [comment]
Please set up codeclimate as well

**Comment by mdyoung3** [comment]
@pookmish I set up Codeclimate.  It appears to fail. I compared the config files to other projects and it looks simpatico. I suspect we need to complete setup before the test will pass. Is that correct, or should I troubleshoot this?

**Comment by pookmish** [comment]
> @pookmish I set up Codeclimate. It appears to fail. I compared the config files to other projects and it looks simpatico. I suspect we need to complete setup before the test will pass. Is that correct, or should I troubleshoot this?

Yeah I see codeclimate is having some issues lately.

Go ahead and fix up the acceptance tests and then this can be merged. But please work on the codeclimate soon.

**Comment by mdyoung3** [comment]
@pookmish Wouldn't the tests--acceptance or otherwise--be the same as ace-gyphon since we haven't made any changes to it?  Or is that it, there things specific to ace-gryphon?

**Comment by pookmish** [comment]
> @pookmish Wouldn't the tests--acceptance or otherwise--be the same as ace-gyphon since we haven't made any changes to it? Or is that it, there things specific to ace-gryphon?

This is the profile. `ace-gryphon` is the stack. You've changed configs and namespaces in this profile. the tests themselves aren't even running at this time. The profile won't even install.

**Comment by mdyoung3** [comment]
I'm more talking about the acceptance testing. For example, the HomePageCest tests for the home page route, looks for text and logs in with admin. 

```  public function testHomepage(AcceptanceTester $I) {
    $I->amOnPage('/');
    $I->canSee('Stanford');
    $I->seeCurrentUrlEquals('/');
    $I->canSee('Welcome to your site!', 'h2');
    $I->logInWithRole('administrator');
    $I->amOnPage('/admin/structure');
    $I->canSeeResponseCodeIs(200);
  }
```

So when you say "fix up the acceptance tests" what do you mean?

**Comment by pookmish** [comment]
> So when you say "fix up the acceptance tests" what do you mean?

https://app.circleci.com/pipelines/github/SU-SWS/trustee_profile/20/workflows/c9bc4ccf-01e7-4edc-8fb7-011653597d32/jobs/40

**Comment by mdyoung3** [comment]
Okay I think I know what you mean now, Mike. I'm seeing the same issue on my local as testing. And `blt doctor` is saying there's config issues.

**Comment by pookmish** [comment]
> Okay I think I know what you mean now, Mike. I'm seeing the same issue on my local as testing. And `blt doctor` is saying there's config issues.

Yes, there's config issues. See the screenshot from the tests:
image

**Comment by mdyoung3** [comment]
Hey @pookmish 

The test are running now, but failing at multiple points. One question I have is, is it possible that anything on the stack side of things is causing this atm?

**Comment by pookmish** [comment]
@mdyoung3, The problem was the CircleCi environment was missing the `STANFORD_ENCRYPT` variable. It just needed to be a 32 character random string. Once I added that and re-ran the tests, everything passed. This is g2g.

**Comment by mdyoung3** [comment]
Yeah, I saw that error. Thanks for taking care of it, @pookmish. Cool. Thanks!

**Pull request merged**